src/FolderContext.ts

```diff
@@ -18,7 +18,7 @@ import { LinuxMain } from "./LinuxMain";
 import { PackageWatcher } from "./PackageWatcher";
 import { SwiftPackage, Target, TargetType } from "./SwiftPackage";
 import { TestExplorer } from "./TestExplorer/TestExplorer";
-import { WorkspaceContext, FolderEvent } from "./WorkspaceContext";
+import { WorkspaceContext, FolderOperation } from "./WorkspaceContext";
```

Comment:
renaming this enum is outside the scope of this PR and creates a lot of noise, especially under src/, let's tackle that as a separate review

Reply:
Unfortunately, it's not just a rename. I had to modify the `WorkspaceContext` to use VS Code Events so that it was testable with `MockUtils`.

Reply:
`type FolderEvent = FolderOperation;`? Just feels like a lot of noise and source of merge conflicts for what's on the go

Reply:
The `FolderEvent` type was replaced with an interface that encapsulates all of the event arguments since `vscode.Event` only accepts a single argument listener. `FolderOperation` is the enumeration that used to be called `FolderEvent`:

```typescript
/** Workspace Folder Operation types */
export enum FolderOperation {
    // Package folder has been added
    add = "add",
    // Package folder has been removed
    remove = "remove",
    // Workspace folder has gained focus via a file inside the folder becoming the actively edited file
    focus = "focus",
    // Workspace folder loses focus because another workspace folder gained it
    unfocus = "unfocus",
    // Package.swift has been updated
    packageUpdated = "packageUpdated",
    // Package.resolved has been updated
    resolvedUpdated = "resolvedUpdated",
}

/** Workspace Folder Event */
export interface FolderEvent {
    operation: FolderOperation;
    workspace: WorkspaceContext;
    folder: FolderContext | null;
}
```

I don't think there's a way to keep backward compatibility here. Changing the name isn't going to work since the listener function will still be different and I don't think making a one-off mock event listener that conforms to the old function really makes sense.

What PR does this conflict with?

Reply:
k, makes sense